db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/SYSCONSTRAINTSRowFactory.java

```diff
@@ -616,4 +618,68 @@ protected int getConstraintType(ExecRow row) throws StandardException{
+                        DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, false, 128),
+                        null,null,view,viewId,0,0,0),
+                new ColumnDescriptor("TBCREATOR",2,2,
+                        DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, false, 128),
+                        null,null,view,viewId,0,0,0),
+                new ColumnDescriptor("TBNAME",3,3,
+                        DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, false, 128),
+                        null,null,view,viewId,0,0,0),
+                new ColumnDescriptor("COLNAME",4,4,
+                        DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, false, 128),
+                        null,null,view,viewId,0,0,0),
+                new ColumnDescriptor("COLSEQ",5,5,
+                        DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.SMALLINT, false),
+                        null,null,view,viewId,0,0,0),
+                new ColumnDescriptor("COLNO",6,6,
+                        DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.SMALLINT, false),
+                        null,null,view,viewId,0,0,0),
+                new ColumnDescriptor("IBMREQD",7,7,
+                        DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.CHAR, false, 1),
+                        null,null,view,viewId,0,0,0),
+                new ColumnDescriptor("PERIOD",8,8,
+                        DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.CHAR, false, 1),
+                        null,null,view,viewId,0,0,0)
+            });
+        return cdsl;
+    }
+
+    public static String SYSKEYCOLUSE_VIEW_IN_SYSIBM = "create view SYSKEYCOLUSE as \n" +
+            "SELECT\n" +
+            "  C.CONSTRAINTNAME AS CONSTNAME,\n" +
+            "  S.SCHEMANAME AS TBCREATOR,\n" +
+            "  T.TABLENAME AS TBNAME,\n" +
+            "  COLS.COLUMNNAME AS COLNAME,\n" +
+            "  CAST(CONGLOMS.DESCRIPTOR.getKeyColumnPosition(COLS.COLUMNNUMBER) AS SMALLINT) AS COLSEQ,\n" +
+            "  CAST(COLS.COLUMNNUMBER AS SMALLINT) AS COLNO,\n" +
+            "  'N' AS IBMREQD, -- IBM release depencency indicators, default 'N'\n" +
+            "  '' AS PERIOD    -- indicates whether the column is the start or end column for the BUSINESS_TIME period, default blank\n" +
+            "FROM\n" +
+            "  (SELECT SC.CONSTRAINTNAME, SC.SCHEMAID, SC.TABLEID, K.CONGLOMERATEID\n" +
+            "     FROM SYS.SYSCONSTRAINTS SC, SYS.SYSKEYS K   -- get all unique constraint conglomerate IDs\n" +
+            "     WHERE K.CONSTRAINTID = SC.CONSTRAINTID\n" +
+            "   UNION\n" +
+            "   SELECT SC.CONSTRAINTNAME, SC.SCHEMAID, SC.TABLEID, PK.CONGLOMERATEID\n" +
+            "     FROM SYS.SYSCONSTRAINTS SC, SYS.SYSPRIMARYKEYS PK\n" +
+            "     WHERE PK.CONSTRAINTID = SC.CONSTRAINTID) C, -- get all primary key conglomerate IDs\n" +
+            "  SYSVW.SYSSCHEMASVIEW S,\n" +
```

Comment:
I think we can use sys.syschemas here instead. I'm considering two reasons: 1) So far, schema restriction hasn't been applied to all the views in sysibm schema yet, so we can be consistent here. 2) the schemasview in your view definition is used only in the second UNION branch, so the schema restriction is not applied in a complete way.

Reply:
Good to know, I change it to `SYS.SYSSCHEMAS` in the latest commit.